static/app/components/events/eventTagsAndScreenshot/index.tsx

```diff
@@ -40,88 +43,110 @@ function EventTagsAndScreenshots({
   }
 
   const showScreenshot = !isShare && !!screenshot;
+  // Check for context bailout condition. No context is rendered if only user is provided
+  const hasEventContext = hasContext && !objectIsEmpty(event.contexts);
   const showTags = !!tags.length || hasContext;
 
   return (
-    <Wrapper
-      isBorderless={isBorderless}
-      showScreenshot={showScreenshot}
-      showTags={showTags}
-    >
+    <Wrapper showScreenshot={showScreenshot} showTags={showTags}>
       {showScreenshot && (
-        <Screenshot
-          organization={organization}
-          event={event}
-          projectSlug={projectSlug}
-          screenshot={screenshot}
-          onDelete={onDeleteScreenshot}
-        />
-      )}
-      {showScreenshot && showTags && <Divider />}
-      {showTags && (
-        <Tags
-          organization={organization}
-          event={event}
-          projectSlug={projectSlug}
-          hasContext={hasContext}
-          location={location}
-        />
+        <ScreenshotWrapper>
+          <Screenshot
+            organization={organization}
+            event={event}
+            projectSlug={projectSlug}
+            screenshot={screenshot}
+            onDelete={onDeleteScreenshot}
+          />
+        </ScreenshotWrapper>
       )}
+      {showScreenshot && showTags && <VerticalDivider />}
+      <TagWrapper hasEventContext={hasEventContext}>
+        {hasEventContext && (
+          <Fragment>
+            <TagsHighlightWrapper showScreenshot={showScreenshot}>
+              <TagsHighlight event={event} />
+            </TagsHighlightWrapper>
+            <Divider />
```

Comment:
what happens if `hasEventContext` equals true and `showTags` false? are we displaying the divider? does it look good?

Reply:
i've blocked this now